ethereum/core/src/main/java/org/hyperledger/besu/ethereum/trie/forest/ForestWorldStateArchive.java

```diff
@@ -66,13 +67,16 @@ public boolean isWorldStateAvailable(final Hash rootHash, final Hash blockHash)
   }
 
   @Override
-  public Optional<MutableWorldState> getMutable(
-      final BlockHeader blockHeader, final boolean isPersistingState) {
-    return getMutable(blockHeader.getStateRoot(), blockHeader.getHash());
+  public Optional<MutableWorldState> getWorldState(final WorldStateQueryParams queryParams) {
+    return getWorldState(queryParams.getStateRoot());
   }
 
   @Override
-  public Optional<MutableWorldState> getMutable(final Hash rootHash, final Hash blockHash) {
+  public MutableWorldState getWorldState() {
```

Comment:
would it be reasonable to rename this to getEmptyWorldState() ?

Reply:
I had a hard time choosing the name for this class because, for Forest, it's the empty state, whereas for Bonsai, it's the head without rollback. So I left it as getWorldState because I couldn't figure out how to align the two. I think once Forest is deprecated, we can rename it or maybe remove it.